Standards/scs-xxxx-v1-ipv4-networking-standard.md

```diff
@@ -0,0 +1,79 @@
+---
+title: IPv4 Networking Standard
+type: Standard
+status: Proposal
+track: IaaS
+---
+
+## Introduction
+
+This document outlines the standardized approach for the management and allocation of
+public IPv4 addresses within Sovereign Cloud Stack (SCS) environments. Its aim is to
+ensure a consistent, secure, and efficient methodology for IP address provisioning
+across all SCS cloud services.
+
+## Motivation
+
+The motivation behind establishing this standard is to enhance interoperability, improve
+security measures, and streamline the operational processes across different SCS clouds.
+It addresses the need for a unified procedure in handling IPv4 networking to facilitate
+ease of use for IaaS users and DevOps teams.
+
+## Design Considerations
+
+### Options considered
+
+#### _Neutron Routers_
+
+Usage of Neutron Routers: To manage traffic between internal and external networks Neutron Routers **MUST** be used as the default gateway for VMs requiring access to external networks and the internet, thereby facilitating the routing of traffic and enhancing network security.
+
+CSPs **SHOULD** use OVN or L3agent as High Availability (HA) service deployments.
```

Comment:
I miss some reasoning why exactly I need to use OVN or L3 agent, also notice that due to some intricacies L3 agent is not really HA currently in some edge cases, like upgrades/reboots, depending on your exact setup - this technical discussion is probably out of scope for this document though.

to be clear I'm totally for using ovn, but we should write something down why we encourage it's use, what are advantages etc.

Reply:
Just to add some background on this point: this is about [high availability of virtual routers](https://access.redhat.com/documentation/de-de/red_hat_openstack_platform/9/html/networking_guide/sec-l3-ha), i.e. replicating virtual routers across multiple network nodes and then allowing failover via VRRP in case of a node failure, a feature that is supported by both virtual router implementations included in neutron (OVN and L3agent), but may be mutually exclusive with distributed virtual routing (DVR, the implementation of virtual routers on the compute nodes).

So this is not really an endorsement of OVN or L3agent, it is just that those are the two available implementations. There might be proprietary service plugins to replace them, but every driver that is not OVN seems to just use L3agent.
 
This feature is also invisible to tenants, and I'm not sure if it should be part of this specific standard. We should probably have a different discussion about where and how to mandate HA features, maybe in the context of #527.